tools/mo/openvino/tools/mo/middle/passes/infer.py

```diff
@@ -220,10 +220,33 @@ def override_batch(graph: Graph, batch: int):
     batch: user defined integer value to override batch
     """
     if batch is not None:
-        for node_id, data in graph.nodes(data=True):
-            if 'op' in data and data['op'] == 'Parameter' and not data.get('fixed_batch', False):
-                validate_batch_in_shape(data['shape'], data['name'])
-                data['shape'][0] = batch
+        in_nodes = graph.get_op_nodes(op='Parameter')
+        layout_values = None
+        if 'layout_values' in graph.graph['cmd_params']:
+            layout_values = graph.graph['cmd_params'].layout_values.copy()
+            if '' in layout_values:
+                if len(in_nodes) == 1:
+                    in_node = in_nodes[0]
+                    layout_values[in_node.soft_get('name', in_node.id)] = layout_values['']
+                    del layout_values['']
+        for node in in_nodes:
+            if not node.soft_get('fixed_batch', False):
+                name = node.soft_get('name', node.id)
+                if layout_values and name in layout_values and layout_values[name]['source_layout']:
+                    from openvino.runtime import Layout  # pylint: disable=no-name-in-module,import-error
+
+                    layout = layout_values[name]['source_layout']
+                    layout_parsed = Layout(layout)
+                    if layout_parsed.has_name('N'):
+                        idx = layout_parsed.get_index_by_name('N')
+                        node['shape'][idx] = batch
```

Comment:
Why we make channel index positive (convert negative to positive) and batch index we don't convert?

Reply:
Here it is obvious that negative index will work as len(shape)-idx, for channels it was done in special function so I decided to normalize index